bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: add configurable CA certificate in Observability RW Proxy OCP template

## Description

Related to https://issues.redhat.com/browse/MGDSTRM-10333.

This PR updates the Observability Remote Write Proxy OCP Template to allow enabling/disabling the usage of a CA certificate to perform TLS verification during the token verification phase in the Observability Remote Write Proxy. During that phase Observability Remote Write Proxy contacts KFM as its token verifier.

This is useful in those scenarios where the token verifier endpoint is an HTTPs endpoint and it uses a self-signed certificate. For example, when referencing a K8s service as the token verifier endpoint  and self-signed TLS certificates  signed by the OCP cluster are demanded for that K8s service.

This PR also creates a K8s ConfigMap where the CA certificate content is automatically set in it with the `service-ca.crt` key. The PR also updates the template to mount that ConfigMap in the Observability Remote Write Proxy pod so it can access it when CA certificate usage is set for the Observability Remote Write Proxy.

## Verification Steps
1. Deploy Observability Remote Write Proxy template. For how to do it see https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/docs/deploying-observability-remote-write-proxy-in-openshift.md. When doing so, set `OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_SERVICE_SERVING_CA_ENABLED` to true
2. Verify that the Observability Remote Write Proxy pod has the `--token.verification.cacert.enabled` is set to true.
3. Verify that a ConfigMap named `observability-remote-write-proxy-service-serving-ca` has been created and it contains a `service-ca.crt` key with content
4. Verify that the Observability Remote Write Proxy pod has a file in it named `/obs-rw-proxy-service-serving-ca/service-ca.crt` with the same content as the value of the ConfigMap mentione above


## Checklist (Definition of Done)
